### ISSUE TITLE
Peer network range posture check also takes into account IP address from disconnected network adapters

### ISSUE THREAD

**Issue by karstennilsen** [netbirdio/netbird#1743]
**Describe the problem**

Peer network range posture check also takes into account IP address from disconnected network adapters.
This is a problem because e.g. on Windows clients WLAN adapters that lose their connection keep their previous DHCP address.

**To Reproduce**

1. Creata a peer network range posture check that blocks: 10.0.31.0/24
2. Assign to a disabled adapter an IP address of 10.0.31.77
3. VPN connectivity get's blocked by posture check

**Expected behavior**

Only connected apapters taken into account. (Maybe even only connected adapters that are used for IP routing?)

**Are you using NetBird Cloud?**

No

**NetBird version**
0.26.3

**Screenshots**

(Verbroken means disconnected in Dutch)
![image_2024-03-22_11-14-55](https://github.com/netbirdio/netbird/assets/8917480/67a70c3c-8e4f-4aff-86a2-7a9af0210969)

**Comment by nazarewk** [comment]
@karstennilsen can you confirm or deny whether it is still an issue?

**Comment by iliaselmatani** [comment]
I can confirm this issue on Windows amd64, NetBird 0.66.4, using NetBird Cloud (`api.netbird.io`).

The affected device has a Peer Network Range posture check configured as a deny list. A physically disconnected Ethernet adapter (cable removed, no `up` or `running` flags) retains its IP address, which falls within the denied range. Even disabling the adapter in Windows Device Manager does not remove the IP from being reported.

**The client correctly identifies the adapter as down for routing:**

```
client/internal/routemanager/systemops/systemops_windows.go:943: interface Ethernet 2 is down, trying next route
client/internal/routemanager/systemops/systemops_windows.go:947: route lookup for 85.x.x.x: selected interface WiFi (index 26), route metric 0, interface metric 35
```

**Yet `localip.go` still includes the ghost IP in every single address report:**

```
client/firewall/uspfilter/localip.go:156: Local IPv4 addresses: [100.x.x.x 169.254.x.x 10.x.x.x 169.254.x.x 172.x.x.x 127.0.0.1]
```

*(The `10.x.x.x` address is from the disconnected adapter; `172.x.x.x` is the active hotspot interface.)*

**This directly causes the peer to be blocked:**

```
client/internal/engine.go:1263: got peers update from Management Service, total peers to connect to = 0
```

This was consistent across multiple client restarts over a ~25-minute window, with the ghost IP present in every `localip.go` entry throughout.

**Disabling the posture check immediately resolved it:**

```
client/internal/engine.go:1263: got peers update from Management Service, total peers to connect to = 2
```

Re-enabling the check blocked the peer again instantly.

The default route and all outbound traffic consistently used the active interface via WiFi, confirming the peer was never actually on the denied network. The posture check logic should filter out interfaces that lack `up` and `running` flags — exactly as the routing logic already does.

**Comment by iliaselmatani** [comment]
> @karstennilsen can you confirm or deny whether it is still an issue?

Yes, this is still an issue for me as well, as described in my comment above. Running NetBird 0.66.4 on Windows amd64, the ghost IP from the disconnected adapter consistently triggers the deny rule.

@nazarewk The "waiting feedback" label can be removed — feedback has been provided confirming this is still an issue (NetBird 0.66.4, Windows amd64, ghost IP from disconnected adapter triggering deny rule). Please add the "bug" label instead.

**Comment by lixmal** [comment]
Fixed by #5768 which skips down interfaces when collecting network addresses for posture checks, released in v0.68.0.
